reana_server/complexity.py

```diff
@@ -0,0 +1,343 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2021 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA workflow complexity estimation."""
+
+import os
+from functools import reduce
+
+from reana_db.models import WorkflowComplexity
+from reana_commons.job_utils import kubernetes_memory_to_bytes
+
+
+def estimate_complexity(workflow_type, reana_yaml):
+    """Estimate complexity in REANA workflow.
+
+    :param workflow_type: A supported workflow specification type.
+    :param reana_yaml: REANA YAML specification.
+    """
+
+    def build_estimator(workflow_type, reana_yaml):
+        if workflow_type == "serial":
+            return SerialComplexityEstimator(reana_yaml)
+        elif workflow_type == "yadage":
+            return YadageComplexityEstimator(reana_yaml)
+        elif workflow_type == "cwl":
+            return CWLComplexityEstimator(reana_yaml)
+        else:
+            raise Exception(
+                "Workflow type '{0}' is not supported".format(workflow_type)
+            )
+
+    estimator = build_estimator(workflow_type, reana_yaml)
+    try:
+        complexity = estimator.estimate_complexity()
+    except Exception:
+        return None
+
+    # Summing up parrarel jobs times memory to determine the workflow complexity
+    memory = reduce(lambda sum, item: sum + item[0] * item[1], complexity, 0)
+
+    # FIXME: Values should be configuranble
```

Comment:
You can check how I'm doing it here to adapt later.

- https://github.com/reanahub/reana/pull/507/files#diff-6846786db1d8b0e9a4b1ff80931cb967630b00a58374384917c9bf356dabdfacR74
- https://github.com/reanahub/reana-db/pull/131/files#diff-9c9ecb5b02eee5df82a9c1138c56e62181e9d8d56152d45c09f7fe0a693ce980R262